### PR TITLE
Issues/#2495 seaborn

### LINKED ISSUE #2495: [upstream?] seaborn 0.10 test issues
**Expected behavior**

All the tests should pass without issues.

**Actual behavior**

`analysis/test_psa.py` fails due to seaborn import issues. This appears to be directly related to the v0.10 release of seaborn, as v0.9 isn't causing the failures.

This appears to currently be the culprit for the recent Travis CI failures.

Trace:

```
self = <MDAnalysisTests.analysis.test_psa.TestPSAnalysis object at 0x7fe0db56f990>, psa = <MDAnalysis.analysis.psa.PSAnalysis object at 0x7fe0dc8965d0>
tmpdir = local('/tmp/pytest-of-mcria/pytest-7/test_dendrogram_produced_annot0')

    @pytest.fixture()
    def plot_annotated_heatmap(self, psa, tmpdir):
        pytest.importorskip('seaborn')
        psa.run(metric='hausdorff')
        with tmpdir.as_cwd():
>           results = psa.plot_annotated_heatmap(filename="annotated.png")

test_psa.py:89:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../package/MDAnalysis/analysis/psa.py:1839: in plot_annotated_heatmap
    None,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

value = None, from_value = None

>   ???
E   ImportError: ERROR --- The seaborn package cannot be found!
E
E                   The seaborn API could not be imported. Please install it first.
E                   You can try installing with pip directly from the
E                   internet:
E
E                     pip install seaborn
E
E                   Alternatively, download the package from
E
E                     http://pypi.python.org/pypi/seaborn/
E
E                   and install in the usual manner.

<string>:3: ImportError
```

**Code to reproduce the behavior**

`pytest test_psa.py`

**Currently version of MDAnalysis**

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 0.20.2-dev0
- Which version of Python (`python -V`)? 3.7.6
- Which operating system? Ubuntu 18.04.3 LTS

### PR DESCRIPTION
**Pull request by RMeli**
Fixes #2495 (temporarily?). 

`seaborn` 0.10 drops support for Python 2.

Changes made in this Pull Request:
 - Cap `seaborn` version to 0.9

PR Checklist
------------
 - [ ] CHANGELOG
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by richardjgowers** [comment]
Not sure this is the right solution, this will filter into pip dependencies
- I don’t want to mess with people’s scientific python stack with a pin.

**Comment by RMeli** [comment]
Should I remove the pin on `setup.py` but leave it on Travis CI? Most of the CI builds are failing because of this problem.

**Comment by richardjgowers** [comment]
Yeah just keep it on travis so we can test for now

**Comment by orbeckst** [comment]
Why can’t conda figure out that the last working version of seaborn for 2 is 0.9? Isn’t this what it’s supposed to be doing?

**Comment by RMeli** [comment]
I don't think the issue is related to Python 2. The failures described in #2495 are obtained with Python 3.x (3.7 for @IAlibay and I confirmed the same problem with 3.6).

**Comment by orbeckst** [comment]
Ah, I misinterpreted the statement "seaborn 0.10 drops support for Python 2." — which I know understand you added to motivate the pin to 0.9.

Thanks for explaining.

File: .travis.yml [hunk 1]
```diff
@@ -28,7 +28,8 @@ env:
     - SETUP_CMD="${PYTEST_FLAGS}"
     - BUILD_CMD="pip install -e package/ && (cd testsuite/ && python setup.py build)"
     - CONDA_MIN_DEPENDENCIES="mmtf-python mock six biopython networkx cython matplotlib scipy griddataformats hypothesis gsd codecov"
-    - CONDA_DEPENDENCIES="${CONDA_MIN_DEPENDENCIES} seaborn>=0.7.0 clustalw=2.1 netcdf4 scikit-learn joblib>=0.12"
+    # Temporarily cap seaborn to 0.9 since 0.10 drops Python 2 support
```
Comment (orbeckst) [inline .travis.yml]: Can you add a link to the actual issue so that, once the issue is resolved, we can remove the pin?

**Comment by RMeli** [comment]
I referenced the issue in my comment. Sorry, I didn't explain clearly that "seaborn 0.10 drops support for Python 2" is an excuse to do cap at 0.9 and get the CI working again for now.

**Comment by orbeckst** [comment]
Thanks @RMeli !

**Pull request merged**